FF96 supports Navigator.canShare()

hamishwillee opened the pull request:
Support added for [Navigator.canShare()](https://developer.mozilla.org/en-US/docs/Web/API/Navigator/canShare) in FF96 according to this bug: https://bugzilla.mozilla.org/show_bug.cgi?id=1666203

- desktop sharing behind preference. Android sharing enabled.
- text shared on desktop, but not on android until this bug fixed https://github.com/mozilla-mobile/fenix/pull/22633
- files still cannot be shared.

makotokato commented on api/Navigator.json:
```diff
@@ -548,10 +548,24 @@
             "firefox_android": {
-              "version_added": false
+              "version_added": "96",
+              "flags": [
```
[dom.webshare.enabled=true](https://searchfox.org/mozilla-central/rev/4c184ca81b28f1ccffbfd08f465709b95bcb4aa1/mobile/android/app/geckoview-prefs.js#37) on Firefox Android as default even if on release channel. It is unnecessary to this `flags` entry. You should remove this `flags` value.

 [Firefox desktop is still false](https://searchfox.org/mozilla-central/rev/4c184ca81b28f1ccffbfd08f465709b95bcb4aa1/modules/libpref/init/StaticPrefList.yaml#4137-4144) on stable channel.

hamishwillee replied:
Thanks very much. I will update this on Friday!

saschanaz commented on api/Navigator.json:
```diff
@@ -548,10 +548,24 @@
+              "flags": [
+                {
+                  "type": "preference",
+                  "name": "#dom.webshare.enabled",
```
Could you remove the leading `#` and change `value_to_set` to `true`? Same for `share`.

hamishwillee commented on api/Navigator.json:
```diff
@@ -657,7 +664,7 @@
               },
               "firefox": {
-                "version_added": false
+                "version_added": "96"
```
FYI @saschanaz I also added that text sharing is supported for share and canshare on desktop in the same versions as they were introduced. I have not added that it is supported for android but we can certainly do so once you merge https://github.com/mozilla-mobile/fenix/pull/22633

hamishwillee commented:
Thanks @saschanaz and @ddbeck. Updated and ready for re-review.

@ddbeck When I removed the prefix # on the preference I noted that there are quite a few others that have this - e.g. `#enable-experimental-web-platform-features`. Are those likely to be typo's too (and if so, do you want me to remove globally)?

saschanaz commented:
Chrome flags can have `#` as they do have preceding sharps, it's just not a Firefox thing.

ddbeck reviewed (approved):
Looks good. Thank you! 🎉
